doc/roadmap.rst

```diff
@@ -0,0 +1,271 @@
+     :issue:`6599` :issue:`6715`
+   * Example of how to develop a meta-estimator
+   * More self-sufficient running of scikit-learn-contrib or a similar resource
+
+#. Support resampling and sample reduction
+
+   * Allow subsampling of majority classes (in a pipeline?) :issue:`3855`
+   * Implement random forests with resampling :issue:`8732`
+
+#. Better interfaces for interactive development
+
+   * __repr__ and HTML visualisations of estimators :issue:`6323`
+   * Include plotting tools, not just as examples. :issue:`9#73`
+
+#. Improved tools for model diagnostics and basic inference
+
+   * partial dependence plots :issue:`5653`
+   * alternative feature importances implementations (e.g. methods or wrappers)
+   * better ways to handle validation sets when fitting
+   * better ways to find thresholds / create decision rules :issue:`86#4`
+
+#. Better tools for selecting hyperparameters with transductive estimators
+
+   * Grid search and cross validation are not applicable to most clustering
+     tasks. Stability-based selection is more relevant.
+
+#. Improved tracking of fitting
+
+   * Verbose is not very friendly and should use a standard logging library
+     :issue:`6929`
+   * Callbacks or a similar system would facilitate logging and early stopping
+
+#. Use scipy BLAS Cython bindings
+
+   * This will make it possible to get rid of our partial copy of suboptimal
+     Atlas C-routines
+   * This should speed up the Windows and Linux wheels
+   * :issue:`#1638`
+
+#. Allow fine-grained parallelism in cython
+
+   * Now that we do not use fork-based multiprocessing in joblib anymore it's
+     possible to use the prange / openmp thread management which makes it
+     possible to have very efficient thread-based parallelism at the Cython
+     level. Example with K-Means: :issue:`#1950`
+
+#. Larger grain (across computers) parallelism
+
+   * Joblib can now plug onto several backends, some of them can distribute the
+     computation across computers
+   * However, we want to stay high level in scikit-learn
+
+#. A way forward for more out of core
+
+   * Dask enables easy out-of-core computation. While the dask model probably
+     cannot be adaptable to all machine-learning algorithms, most machine
+     learning is on smaller data than ETL, hence we can maybe adapt to very
+     large scale while supporting only a fraction of the patterns.
+
+#. Better support for manual and automatic pipeline building
+
+   * Easier way to construct complex pipelines and valid search spaces
+     :issue:`7608` :issue:`5082` :issue:`8243`
+   * provide search ranges for common estimators??
+   * cf. `searchgrid <https://searchgrid.readthedocs.io/en/latest/>`_
+
+#. Support for working with pre-trained models
+
+   * Estimator "freezing". In particular right now it's impossible to clone a
+     CalibratedClassifierCV with prefit. :issue:`8370`. :issue:`645#`
+
+#. Backwards-compatible de/serialization of some estimators

```

Comment:
This seems to be a sub-point of the thing below, right? but maybe we can leave it like that for now?